scripts/build_shapes.py

```diff
@@ -80,7 +82,61 @@ def download_GADM(country_code, update=False, out_logging=False):
     return GADM_inputfile_gpkg, GADM_filename
 
 
-def get_GADM_layer(country_list, layer_id, geo_crs, update=False, outlogging=False):
+def filter_gadm(
+    geodf,
+    layer,
+    cc,
+    contended_flag,
+    output_nonstd_to_csv=False,
+):
+
+    # identify non standard geodf rows
+    geodf_non_std = geodf[geodf["GID_0"] != two_2_three_digits_country(cc)].copy()
+
+    if not geodf_non_std.empty:
+        logger.info(
+            f"Contended areas have been found for gadm layer {layer}. They will be dropped according to {contended_flag} option"
+        )
+
+        # NOTE: in these options GID_0 is not changed because it is modified below
+        if contended_flag == "drop":
+            geodf.drop(geodf_non_std.index, inplace=True)
+        else:
+            # this is the "set_by_country" option that is also the fallback for unexpected values
+            if contended_flag != "set_by_country":
+                logger.warning(
+                    f"Value '{contended_flag}' for option contented_flag is not recognized.\n"
+                    + "Fallback to 'set_by_country'"
+                )
+
+    # force GID_0 to be the country code for the relevant countries
+    geodf["GID_0"] = cc
+
+    # country shape should have a single geomerty
+    if (layer == 0) and (geodf.shape[0] > 1):
+        logger.warning(
+            f"Country shape contains multiple contended areas that are being merged in agreement to contented_flag option '{contended_flag}'"
```

Comment:
Can we change the level of the condition to be sure that merging is done if and only if there are non-standard codes?

Reply:
I feel like we want that to happen even when there are not standard zones.
However, it is important to notify that.
Anyway, this condition would only apply to country shapes so I think it is fine